src/converse-chatview.js

```diff
@@ -855,6 +856,37 @@ converse.plugins.add('converse-chatview', {
                 }
             },
 
+            onMuteUserError (error) {
+                console.error('Stanza error:', error);
+            },
+
+            onMuteUserSuccess (jid) {
+                const messages = this.model.messages;
+                const removeMessage = message => {
+                    const from = message.get('from');
+                    if (from === jid) message.destroy();
+                };
+                messages.models.forEach(removeMessage);
```

Comment:
Done. However, it seems that for some reason the `destroy` method, even though it is called the right number of times, doesn't really destroy the number messages that it should. Any thoughts on this?

Reply:
It might be because the collection's length changes as it's looping through it.

Try using `Array.from` and mapping that to a list of promises that you wait for, like is done here:

https://github.com/conversejs/skeletor/blob/b260c554f4ce961c29deea4740083e58a489aa9b/src/collection.js#L233